components/Builder.tsx

```diff
@@ -94,11 +99,13 @@ export default function Builder({ data }: { data: any }) {
   };
 
   const formationRef = createRef<HTMLDivElement>();
+  const searchInputRef = createRef<HTMLInputElement>();
 
   const { isSignedIn, user } = useUser();
 
   function updateFormation(slot: number, character: Character) {
     setFormation(updateSlotInFormation(formation, slot, character));
+    setCharacterFilter((prev) => ({ ...prev, name: "" }));
```

Comment:
this seems odd to me, when updating the formation you want to clear the search bar?

Reply:
If you search a unit, eg. Parisa, you add the unit to the formation. You're then left with an empty hero grid until you clear the formation so clearing the grid allows you to move onto the next hero faster.

Reply:
got it, i guess it is safe to assume that after typing into the search bar the user will be adding that character the vast majority of the time. 

i was thinking about the case where you had some test in the search bar but then moved a character in the layout first. that would also clear the search bar, but i doubt that's nearly as common of a situation.

LGTM